Explicit executable for running 'easy_install' and 'pip'.

paparomeo opened the pull request:
New `executable` module argument for `easy_install` and `pip` packaging modules, that allows specifying explicitly the executable to use for running `easy_install` and `pip`, allowing for example, the system installation of packages for a specific version of _Python_ when there are multiple versions of _Python_ installed in the system.

mpdehaan commented:
Flask author has also said unkind things about Ansible for being GPL, so let's use another example than Flask.  (Yes, I'm serious :))

mpdehaan commented:
We should probably name the argument "executable" to align with other modules and not be somewhat confusing as there is already a module named 'command'.

mpdehaan commented:
Also strange the binary is not pathed explicitly but rather a directory above it containing bin.

If specified, we should use the binary name verbatim.

Commit:
Changes examples from using Flask to some other package (Bottle was chosen) as per @mpdehaan's suggestion.

Commit:
Rename copy/pasted variables more appropriately.

Commit:
New module argument to specify the executable used for running 'easy_install'. This allows support for system installation of packages on systems with multiple installations of Python.

Commit:
New module argument to specify the executable used for running 'pip'. This allows support for system installation of packages on systems with multiple installations of Python.

paparomeo commented:
@mpdehaan: refactored the pull request according to your comments. Hope _bottle_ is an acceptable replacement for _flask_ in the examples.

Regarding the `bin/` prefix comment, you'll notice that it is only used for the code that is handling the _virtualenv_ cases. It was already there prior to my changes and I wanted my changes to be as least intrusive as possible.

Commit:
Refactor helper function '_get_pip' to handle the cases where an absolute
path explicit executable is passed to the task and to look for an explicit executable
by basename in a virtualenv if that is specified.

Commit:
Implement a '_get_easy_install' helper function to find the desired
'easy_install' executable, modeled on '_get_pip'
from the 'pip' module to correctly handle all combinations of explicit
executable and virtualenv.

paparomeo commented:
@jimi1283: pull request updated with your suggestions.

jimi-c commented:
Thanks, I'll take a look at this today.

ariddell commented:
This would likely fix my bug report in #4253

paparomeo commented:
@jimi-c: ping? Is there some technical shortcoming of the pull request preventing it from being accepted?

jimi-c commented:
@pmcnr No technical reason, there's just a backlog of community submissions we've been working through. Sorry for the delay in getting to this one.

paparomeo commented:
@jimi-c: no worries. Thanks for the feedback!

mpdehaan commented:
merged in, thanks!

Appreciate your patience with the delay on the backlog!

paparomeo commented:
No worries. Happy you've accepted the contribution.
